keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -450,157 +488,101 @@ about permitting exports from `kube-system` and `default`. As a best practice,
+multi-cluster service. _An implementation may or may not decide to create
+missing namespaces automatically, that behavior is out of scope of this spec._
 
 For each exported service, one `ServiceExport` will exist in each cluster that
-runs the service. The mcsd-controller will create and maintain a derived
-`ServiceImport` in each cluster within the supercluster (see: [constraints and
-conflict resolution](#constraints-and-conflict-resolution)). If all `ServiceExport` instances are deleted, each
-`ServiceImport` will also be deleted from all clusters.
+exports the service. The mcs-controller will create and maintain a derived
+`ServiceImport` in each cluster within the supercluster so long as the service's
+namespace exists (see: [constraints and conflict
+resolution](#constraints-and-conflict-resolution)). If all `ServiceExport`
+instances are deleted, each `ServiceImport` will also be deleted from all
+clusters.
 
 Since a given `ServiceImport` may be backed by multiple `EndpointSlices`, a
 given `EndpointSlice` will reference its `ServiceImport` using the label
-`multicluster.kubernetes.io/imported-service-name` similarly to how an
-`EndpointSlice` is associated with its `Service` in a single cluster. Each
-imported `EndpointSlice` will also have a
-`multicluster.kubernetes.io/source-cluster` label with a registry-scoped unique
-identifier for the cluster.
+`multicluster.kubernetes.io/service-name` similarly to how an
+`EndpointSlice` is associated with its `Service` in a single cluster.
+
+Each imported `EndpointSlice` will also have a
+`multicluster.kubernetes.io/source-cluster` label with the cluster name, a
+registry-scoped unique identifier for the cluster. The `EndpointSlice`s imported
+for a service are not guaranteed to exactly match the originally exported
+`EndpointSlice`s, but each slice is guaranteed to map only to a single source
+cluster.
+
+The mcs-controller is responsible for managing imported `EndpointSlice`s.
 
 ```golang
-// ServiceImport declares that the specified service should be exported to other clusters.
+// ServiceImport describes a service imported from clusters in a supercluster.
 type ServiceImport struct {
- metav1.TypeMeta `json:",inline"`
- // +optional
- metav1.ObjectMeta `json:"metadata,omitempty"`
- // +optional
- Spec ServiceImportSpec `json:"spec,omitempty"`
+  metav1.TypeMeta `json:",inline"`
+  // +optional
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+  // +optional
+  Spec ServiceImportSpec `json:"spec,omitempty"`
+  // +optional
+  Status ServiceImportStatus `json:"status,omitempty"`
 }
 
-// ServiceImportSpec contains the current status of an imported service and the
-// information necessary to consume it
+// ServiceImportType designates the type of a ServiceImport
+type ServiceImportType string
+
+const (
+  // SuperclusterIP are only accessible via the Supercluster IP.
+  SuperclusterIP ServiceImportType = "SuperclusterIP"
```

Comment:
I had that on in my deck on Tuesday but we ran out of time. I'll add it for next week

Reply:
Do you want to start a spreadsheet and share it, and we can gather  suggestions (brainstorm-style, no judgement) ?

Reply:
do we actually need a ServiceImportType? can ```ServiceImport.Spec.IP``` have the same semantics as ```Service.Spec.ClusterIP``` (i.e., ```None``` means headless)?

Reply:
Explicit seems to be a safer choice. If we could start over with `Service` we may have wanted `Headless` to be a separate type there too since it's a fundamentally different kind of service.

Reply:
@ahg-g I wrote that comment and then deleted it for the reason @JeremyOT cites.  "None" is a mess.  In this context, we don't even need "None" - we could say "" means headless, but that doesn't leave any room for growth and again, what @JeremyOT says - "headless" should probably have been a `type`.  Now, I also hate the word "headless" but it's more or less part of the vernacular now.